src/app/state/rewardSlice.ts

```diff
@@ -32,8 +33,14 @@ interface RewardConfig {
   };
 }
 
-function findFieldByName(fieldName: string, fields: any[]) {
-  return fields.find((field) => field.field_name === fieldName);
+function findFieldValueByNameOrThrow(fieldName: string, fields: any[]): string {
```

Comment:
This function should throw an error, as getting the component addresses is a mandatory step. Without it we won't be able to fetch rewards or claim, so better to throw error and handle the error accordingly.